internal/slack/slack.go

```diff
@@ -368,3 +371,36 @@ func (c *Client) NotifyK8SPodErrorEvent(ctx context.Context, event *http.PodErro
 	}
 	return err
 }
+
+func (c *Client) NotifyReleaseManagerError(ctx context.Context, msgType, service, environment, branch, namespace, authorEmail string, err error) error {
+	if c.muteOptions.ReleaseManagerError {
+		return nil
+	}
+	userID, err := c.getIdByEmail(ctx, authorEmail)
+	if err != nil {
+		// If user id somehow couldn't be found, post the message to #squad-nasa
+		userID = "#squad-nasa"
+	}
+
+	asUser := slack.MsgOptionAsUser(true)
+	attachments := slack.MsgOptionAttachments(slack.Attachment{
+		Title:      fmt.Sprintf(":boom: Release Manager failed :x:"),
+		Color:      MsgColorRed,
+		Text:       generateSlackMessage(msgType, service, environment, branch, namespace, err),
+		MarkdownIn: []string{"text", "fields"},
+	})
+	_, _, err = c.client.PostMessageContext(ctx, userID, asUser, attachments)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func generateSlackMessage(msgType, service, environment, branch, namespace string, err error) string {
+	switch {
+	case msgType == "promote" && service != "" && environment != "" && branch != "":
+		return fmt.Sprintf("Failed promoting %s #%s in %s. Try promiting again.\nError: %s", service, branch, environment, err)
```

Comment:
We could add the command for them to copy paste and highlight as code, maybe? 

Try to run the command again: `hamctl promote ...`

Reply:
Definitely. What if the info is malformed though? Most likely there is something wrong with the input, since it broke in the first place, right? Otherwise we could just requeue.

Reply:
Actually, that is a great point. It is not due to input as we have fixed that condition in the syncronous step. The errros we’ve seen since are memory limits and alike. Maybe we should just requeue?

Reply:
If we requeue, we should only do it a few times, then fallback to above behavior. Do we have possible counter to increment?

Reply:
Hmm, we can use the redeliver property, to ensure it doesn't continue forever, but only try once. Or we can send it back: https://stackoverflow.com/questions/23158310/how-do-i-set-a-number-of-retry-attempts-in-rabbitmq

Reply:
Yes, we can setup some limitations for it, but that gets more complicated.
Let's just stick with the `Nack` right now to keep it simple.

Reply:
In order to let the `msgType` propagate into this handler, we could use a custom `error` implementation. That way the handlers can decide what the message should be and we can avoid the generic event?

Reply:
That's definitely also a way to go @Crevil. Does that mean each error handler would have to implement some interface? And then fall back to a generic one? 

I implemented the redeliver once, with the `msg.Redelivered` prop, for retry once behavior.